### ISSUE TITLE
Ability to change base path for deployment

### ISSUE THREAD

**Issue by msharara1998** [vmware/data-annotator-for-machine-learning#346]
### Is your feature request related to a problem? Please describe.

Yes, I want to deploy the app and the annotation service behind an nginx reverse proxy under a base path `/annotate-service`.

### Describe the solution you'd like

Ability to change the base path from an environment variable.

**Comment by Kai-Qiao** [comment]
Hi, @msharara1998 want to make the requirements clear with you.
my understanding is you want to deploy _**DAML**_ application on your server, let's assume the server domain is `annotation.service.com`, with the domain you can open the application web page, and then you just want the API path (annotation-service application URL) to be `annotation.service.com/annotate-service/api/v1.0/xxxx` ? (`/api/v1.0/xxx` is the original URL)

**Comment by Kai-Qiao** [comment]
Hi @msharara1998  is the comments meet with your thought?

**Comment by msharara1998** [comment]
yes as you suggested, we want to deploy under something like `[https://annotation.service.com/annotate-service/`](https://annotation.service.com/annotate-service/%60)

**Comment by Kai-Qiao** [comment]
Hi @msharara1998 
as said assume your domain is `annotation.service.com`
you just need to update **annotationService** to `http://annotation.service.com/annotate-service` at https://github.com/vmware/data-annotator-for-machine-learning/blob/master/annotation-app/src/environments/environment.ts#L15

- [1 ] For annotation-service Nginx config as below

  ```
  location  /annotation-service/ {
              proxy_pass http://localhost:3000/;
         }
  ```

- [2 ] There have two ways to config Nginx for the annotation-app you can choose one to use

  - deploy annotation-app build result to Nginx html forld
    1. cd annotation-app/
    2. npm run build
    3. copy the files at dist/ to Nginx html/
    4. Nginx config as below:
      ```     
      location  / {
                  root   html;
                  index  index.html index.htm;
                  try_files $uri $uri/ /index.html =404;
             }
      ```
  
  - run annotation-app as a service
    1. cd annotation-app/
    2. npm run start
    3. Nginx config as below:
      ```
      location  / {
                  proxy_pass http://localhost:4200/;
             }
      ```

**Comment by Kai-Qiao** [comment]
The complete Nginx configuration:
```
server {
       listen       80;
       server_name  annotation.service.com;
       
       underscores_in_headers on; 
       charset utf-8;
       add_header Access-Control-Allow-Origin $http_origin;
       add_header Access-Control-Allow-Headers "Authorization";
       add_header Access-Control-Allow-Methods *;
       proxy_redirect off;
       proxy_http_version 1.1;
       proxy_set_header Upgrade $http_upgrade;
       proxy_set_header Connection "upgrade";
       proxy_set_header Host $proxy_host;
       proxy_set_header X-Real-IP  $remote_addr;
       proxy_set_header X-Forwarded-For $remote_addr;
       proxy_set_header Host $host;
       proxy_set_header Authorization $http_authorization;
       proxy_pass_header Authorization;
       proxy_set_header "Connection" "";
       proxy_pass_request_headers on;
       
       location  / {
            proxy_pass http://localhost:4200/;
       }

    #     location  / {
    #         root   html;
    #         index  index.html index.htm;
    #         try_files $uri $uri/ /index.html =404;
    #    }

       location  /annotation-service/ {
            proxy_pass http://localhost:3000/;
       }
    
}
```

**Comment by Kai-Qiao** [comment]
For local use or don't use Nginx and want to add the base path need update the below files:

- annotation-app:  https://github.com/vmware/data-annotator-for-machine-learning/blob/master/annotation-app/src/environments/environment.ts
`annotationService: 'http://localhost:3000/basepath'`

- annotation-service: https://github.com/vmware/data-annotator-for-machine-learning/blob/master/annotation-service/config/constant.js
`API_BASE_PATH: process.env.API_BASE_PATH || '/basepath',`

**Comment by Kai-Qiao** [comment]
And remember if you are enabled to send email notification functions you should still update the right annotation-app and annotation-service URL at: https://github.com/vmware/data-annotator-for-machine-learning/blob/master/annotation-service/config/app-os.js
```
  WebClientUrl: process.env.WEBCLIENT_URL || 'http://localhost:4200',
  annotationServiceUrl: process.env.ANNOTATION_SERVICE_URL || 'http://localhost:3000',
```

**Comment by msharara1998** [comment]
Okay, deal
thank you man!
